swaybar/render.c

```diff
@@ -14,55 +14,73 @@
 #include "swaybar/status_line.h"
 #include "wlr-layer-shell-unstable-v1-client-protocol.h"
 
-static const int ws_horizontal_padding = 5;
-static const double ws_vertical_padding = 1.5;
-static const double border_width = 1;
+static const int WS_HORIZONTAL_PADDING = 5;
+static const double WS_VERTICAL_PADDING = 1.5;
+static const double BORDER_WIDTH = 1;
 
 static uint32_t render_status_line_error(cairo_t *cairo,
-		struct swaybar_config *config, const char *error,
-		double *x, uint32_t width, uint32_t height) {
+		struct swaybar_output *output, struct swaybar_config *config,
+		const char *error, double *x, uint32_t width, uint32_t surface_height) {
 	if (!error) {
 		return 0;
 	}
+
+	uint32_t height = surface_height * output->scale;
+
 	cairo_set_source_u32(cairo, 0xFF0000FF);
-	static const int margin = 3;
+
+	int margin = 3 * output->scale;
+	int ws_vertical_padding = WS_VERTICAL_PADDING * output->scale;
+
 	int text_width, text_height;
 	get_text_size(cairo, config->font,
-			&text_width, &text_height, 1, false, "%s", error);
+			&text_width, &text_height, output->scale, false, "%s", error);
+
 	uint32_t ideal_height = text_height + ws_vertical_padding * 2;
 	if (height < ideal_height) {
-		return ideal_height;
+		return ideal_height / output->scale;
 	}
 	*x -= text_width + margin;
+
 	double text_y = height / 2.0 - text_height / 2.0;
 	cairo_move_to(cairo, *x, (int)floor(text_y));
-	pango_printf(cairo, config->font, 1, false, "%s", error);
+	pango_printf(cairo, config->font, output->scale, false, "%s", error);
 	*x -= margin;
-	return ideal_height;
+	return ideal_height / output->scale;
 }
 
 static uint32_t render_status_line_text(cairo_t *cairo,
-		struct swaybar_config *config, const char *text,
-		bool focused, double *x, uint32_t width, uint32_t height) {
+		struct swaybar_output *output, struct swaybar_config *config,
+		const char *text, bool focused, double *x,
+		uint32_t width, uint32_t surface_height) {
```

Comment:
`surface_width`

Reply:
It's unused actually, does it make sense to remove it?

Reply:
yup